### ISSUE TITLE
Feature Request: OTEL Collector Operator - Multiple Configuration Support

### ISSUE THREAD

**Issue by vidyakamal** [open-telemetry/opentelemetry-operator#2079]
Requesting support for multiple configurations in the OTEL Collector Operator. While the OTEL Collector itself allows  multiple config files - https://github.com/open-telemetry/opentelemetry-collector/pull/4727,  the Operator currently supports a 1:1 CR to Deployment mapping. This enhancement would offer more deployment flexibility and convenience for managing the OTEL Collector instances.

**Comment by vidyakamal** [comment]
Yes, if the answer to this question is 'YES" -> 'What would be the behavior when the internal content of these specified ConfigMaps is updated? Will OTEL Operator be watching for updates on these ConfigMaps and restart Collectors in order to pickup new changes?'

**Comment by jaronoff97** [comment]
See @swiatekm-sumo's comment [here](https://github.com/open-telemetry/opentelemetry-operator/issues/2070#issuecomment-1707900707), TL;DR no, we would recommend a strategy where you hash the configmap and set it as a pod annotation.
